tests/cpp/data/test_sparse_page_dmatrix.cc

```diff
@@ -51,14 +51,14 @@ TEST(SparsePageDMatrix, ColAccess) {
   EXPECT_EQ(dmat->GetColDensity(1), 0.5);
 
   // Loop over the batches and assert the data is as expected
-  for (auto col_batch : dmat->GetBatches<xgboost::SortedCSCPage>()) {
+  for (auto const& col_batch : dmat->GetBatches<xgboost::SortedCSCPage>()) {
```

Comment:
I wonder if we can delete the copy constructor to prevent this in future.

Reply:
Yes.  I tried to delete the copy constructor of Host device vector and found these.  Will get a PR with another fix.